common/core/nesteddisplay.h

```diff
@@ -17,20 +17,103 @@
 #ifndef COMMON_CORE_NESTEDDISPLAY_H_
 #define COMMON_CORE_NESTEDDISPLAY_H_
 
+#include <nativedisplay.h>
 #include <stdlib.h>
 #include <stdint.h>
 
+#include <linux/hyper_dmabuf.h>
 #include <memory>
-
+#include <map>
+#include "drmbuffer.h"
+#include <utils/threads.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+
+#include "hwcthread.h"
+#include "hwctrace.h"
 #include <nativedisplay.h>
+#define SURFACE_NAME_LENGTH    64
 
 namespace hwcomposer {
 
+struct HwcLayer;
+class NativeBufferHandler;
 class NestedDisplayManager;
 
+struct vm_header {
+  int32_t version;
+  int32_t output;
+  int32_t counter;
+  int32_t n_buffers;
+  int32_t disp_w;
+  int32_t disp_h;
+};
+
+struct vm_buffer_info {
+  int32_t width, height;
+  int32_t format;
+  int32_t pitch[3];
+  int32_t offset[3];
+  int32_t bpp;
+  int32_t tile_format;
+  int32_t rotation;
+  int32_t status;
+  int32_t counter;
+  union {
+    hyper_dmabuf_id_t hyper_dmabuf_id;
+    unsigned long ggtt_offset;
+  };
+  char surface_name[SURFACE_NAME_LENGTH];
+  uint64_t surface_id;
+  int32_t bbox[4];
+};
+
+class SocketThread: public HWCThread {
+
+public:
+  SocketThread( int* client,bool* connection, int server):HWCThread(-8, "SocketThread") {
```

Comment:
Move the implementation to cpp file

Reply:
you means the socketthread functions?

Reply:
ya define the interface in header and move the implementation to cpp

Reply:
ok, will do it.